components/brave_rewards/common/mojom/rewards_page.mojom

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+module brave_rewards.mojom;
+
+import "brave/components/brave_rewards/common/mojom/rewards.mojom";
+
+// Initializes messaging between the browser and the page.
+interface RewardsPageHandlerFactory {
+  CreatePageHandler(pending_remote<RewardsPage> page,
+                    pending_receiver<RewardsPageHandler> receiver);
+};
+
+// Page-side handler for notifications from the browser.
+interface RewardsPage {
+  // Notifies the page that Rewards state has changed and updated information
+  // should be fetched.
+  OnRewardsStateUpdated();
+};
+
+struct AvailableCountryInfo {
+  array<string> country_codes;
+  string default_country_code;
+};
+
+// Browser-side handler for requests from WebUI page.
+interface RewardsPageHandler {
+  // Notifies the browser that that Rewards page has rendered and is ready to be
+  // displayed to the user.
+  OnPageReady();
+
+  // Opens a new tab with the specified URL.
+  OpenTab(string url);
```

Comment:
Is this exposed to the every page?

Reply:
Why we need another OpenURL implementation? Isn't `window.open` enough?

Reply:
When this webUI is displayed as a bubble (i.e. as the Rewards panel), `window.open` does not work. The old Rewards panel uses `chrome.tabs.create` to open tabs, but I assume this is preferred.

It allows any URL to be opened, since eventually we'll need to open links to Rewards "creator" sites.

Reply:
It might be safer to use the already exposed JS methods here. WDYT @bridiver ?